great_expectations/datasource/new_datasource.py

```diff
@@ -17,7 +18,7 @@
 logger = logging.getLogger(__name__)
 
 
-class BaseDatasource:
+class BaseDatasource(object):
```

Comment:
More important: why is this "BaseDatasource" instead of Datasource -> I think Base should be reserved for an abstract class, but I think this *is* meant to be instantiated.

Less important:
I just had to look this up because I've been curious about this, but I think the current python docs no longer recommend specifying `object` as a base class.

Reply:
@jcampbell I was not awre that `object` is no longer required; thank you -- so I removed it.

As far as `BaseDatasource` -- we made it `BaseDatasource` during the batchpod work (this is not new in the present PR).  For what it is worth, `BaseDatasource` is never instantiated; it is separated to provide common methods to `Datasource` (which is the new `Datasource` as compared to `LegacyDatasource`) and to `SimpleSqlalchemyDatasource` (which has significant differences from files-focused datasource classes, hence on its separate inheritance edge).